Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
I'd leave HEAD as HEAD; I think the most predictable things to do with HEAD are upgrade always or never, and probably "never" because I've installed llvm with --HEAD before :) I think it's reasonable for brew to abandon responsibility for upgrading formulas installed from head and rely on the judgement of the user.